doc/ug/getting_started_build_sw.md

```diff
@@ -89,12 +90,139 @@ You can still use `TOOLCHAIN_PATH` to point to the toolchain location if you hav
+
+Specific targets can be built with:
+
+```console
+bazel build //<path to build file>:<name of target>
+```
+
+### Structure of our Bazel workspace
+
+The rules for Bazel are described in a language called starlark, which looks a lot like Python.
+
+The OpenTitan directory is defined as a Bazel workspace by the ```//WORKSPACE``` file.
+``BUILD``` files provide the information Bazel needs to build the targets in a directory.
+```BUILD``` files also manage any subdirectories that don't have their own ```BUILD``` files.
+
+OpenTitan uses .bzl files to specify rules to build artifacts that require specific attention like target specific test rules and project specific binaries.
+
+#### WORKSPACE file
+
+The ```WORKSPACE``` file controls external dependencies such that builds can be made reproducible and hermetic.
+Bazel loads specific commits of external git repositories.
+It uses them to build OpenTitan targets (like it does with bazel\_embedded) or to satisfy dependencies (as it does with abseil).
+To produce increasingly stable releases the ```WORKSPACE``` file will fix a number of these references to specific commits and not the default branch head.
+Dependencies will be added to the workspace so builds and tests are less sensitive to external updates and have a simpler getting started flow.
+
+#### BUILD files
+
+Throughout the OpenTitan repo, ```BUILD``` files describe targets and dependencies in the same directory and subdirectories that lack their own ```BUILD``` files.
+```BUILD``` files are mostly hand-written; to suit the request that hand-written files not be included in autogen directories, there are large ```BUILD``` files that describe how to build and depend on auto-generated files in autogen subdirectories.
+
+### Linting Starlark
+
+The Linter for Bazel files is called "buildifier" and can be built and run with Bazel by running:
+
+```console
+bazel run //buildifier_fix
+```
+
+Which will run the linter on all of the WORKSPACE,BUILD and .bzl files in the workspace.
+It's fairly strict so the best way to keep git logs and blame reports clean is to run it before committing, and ask the same during code review.
+
+### Test Execution
+
+Bazel will execute tests when invoked with ```bazel test <label-expression>```.
+The test rules have various tags applied to them to allow for filtering of tests during dependency analysis.
+Our tests are typically tagged with verilator, cw310, or manual.
+The ```--test_tag_filters``` switch can be used to filter out tests that you either cannot or do not want to execute.
+
+For example, if you do not have a CW310 FPGA board, you cannot execute the tests meant to execute on that board.
+You can instruct Bazel to not execute those tests:
+
+```console
+bazel test --test_tag_filters=-cw310 //...
+```
+
+Rather than manually adding that command line switch to every invocation of bazel, you can add it to your $HOME/.bazelrc file.
+
+
+$HOME/.bazelrc:
```

Comment:
I think this needs to be wrapped in a code block.

Reply:
Yeah, I was wary of mixing the path specification with the code, but you're right and I've already described the path inline so I'll use that